utils/check_config_attributes.py

```diff
@@ -0,0 +1,179 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib
+import inspect
+import os
+import re
+
+
+# All paths are set with the intent you should run this script from the root of the repo with the command
+# python utils/check_config_docstrings.py
+PATH_TO_TRANSFORMERS = "src/transformers"
+
+
+# This is to make sure the transformers module imported is the one in the repo.
+spec = importlib.util.spec_from_file_location(
+    "transformers",
+    os.path.join(PATH_TO_TRANSFORMERS, "__init__.py"),
+    submodule_search_locations=[PATH_TO_TRANSFORMERS],
+)
+transformers = spec.loader.load_module()
+
+CONFIG_MAPPING = transformers.models.auto.configuration_auto.CONFIG_MAPPING
+
+CASES_TO_ALLOW = {
+    "BioGptConfig": ["layer_norm_eps"],
+    "CvtConfig": ["layer_norm_eps"],
+    "DPRConfig": ["layer_norm_eps"],
+    "GLPNConfig": ["layer_norm_eps"],
+    "PerceiverConfig": ["layer_norm_eps"],
+    "RetriBertConfig": ["layer_norm_eps"],
+    "SegformerConfig": ["layer_norm_eps"],
+    "TrajectoryTransformerConfig": ["layer_norm_eps"],
+}
+
+
+def check_attribute_being_used(config_class, attributes, default_value, modeling_sources):
+    """Check if any name in `attributes` is used in one of the strings in`modeling_sources`"""
```

Comment:
```suggestion
    """Check if any name in `attributes` is used in one of the strings in `modeling_sources`"""
```
Would be nice to also explain `config_class` and `default_value` here :-)